commands/box/contract/root.go

```diff
@@ -1226,16 +1226,15 @@ func compileSol(filepath string) (binData string, abiData string, err error) {
 	cmd := exec.Command("solc", filepath, "--bin", "--abi")
 	//CombinedOutput runs the command and returns its combined standard output and standard error.
 	output, err := cmd.CombinedOutput()
-	outputStr := string(output)
-	//an error is returned and the error message is the output.
 	if err != nil {
-		return "", "", errors.New(outputStr)
+		//an error is returned and the error message is the output.
+		return "", "", errors.New(string(output))
 	}
+	outputStr := string(output)
 	binMatches := solcBinReg.FindStringSubmatch(outputStr)
 	abiMathes := solcAbiReg.FindStringSubmatch(outputStr)
 	if len(abiMathes) > 0 {
```

Comment:
here the length of abiMathes is 1 at least
you should use "len(abiMathes) == 2"